packages/fbtee/src/list.tsx

```diff
@@ -0,0 +1,134 @@
+/// <reference types="../ReactTypes.d.ts" />
+
+import invariant from 'invariant';
+import { isValidElement, ReactElement, ReactNode } from 'react';
+import fbt from './fbt.tsx';
+
+export type Conjunction = 'and' | 'none' | 'or';
+export type Delimiter = 'bullet' | 'comma' | 'semicolon';
+
+export default function list(
+  items: ReadonlyArray<string | ReactElement | null | undefined>,
```

Comment:
Yes… the only problem is that TypeScript doesn't have opaque types 🫠

Reply:
Back then the `Fbt` type had to include the `string` type because of all the legacy code, but as a standalone library, fbtee could afford to remove it. 
Which would effectively mean we can finally consider `Fbt` a properly translated string after all. 
Would that be a good alternative?

Reply:
If we remove `string` as the result of `fbt()`, how would you turn it into a string then that TypeScript would be happy about?

In general, I appreciate any cleanups and tightening of the types/API. Ideally in the end we'll have something like:
* `fbt()` terminates into a string that can be inserted anywhere strings are allowed (including React).
* `<fbt>` is a real React component and cannot be used as a string, like for example via `alert(<fbt>)`.

btw. I did add a kind of "opaque"  `TranslatedString` type that works for me in Athena Crisis: https://github.com/nkzw-tech/fbtee/blob/main/packages/fbtee/src/Types.d.ts#L191

Reply:
Actually, the result from fbt() or <fbt> is not meant to be a true string replacement, because String prototype methods aren't i18n-safe. E.g. .substring(), .toUpperCase(), etc... don't handle unicode characters properly from an i18n POV. (Multi byte characters can get trimmed incorrectly, capitalization only works in the basic alphabet). In general, everytime some code takes an fbt result and tries to edit or concatenate it to another string is considered code smell.

Because Fbt was invented in the context of Facebook's massive legacy code base, we attached some stringish methods to keep things going... But it doesn't have to always be like that.
We could finally make Fbt a more traditional JS class instance with only the .toString() method - although that method should be mainly used by an UI library like React instead of userland code. 

Internally, "Fbt result" (IIRC FbtResultBase) was the "terminating" type and was added
 to the ReactNode type definition. 

----

> fbt() terminates into a string that can be inserted anywhere strings are allowed (including React).

I agree with your overall vision. To talk in terms of types, I hope we could make the Fbt type become a pure string (which is currently known as Fbs), and ReactFbt would be the JSX-version of it that may contain nested react elements.